src/main/java/org/logstash/plugins/inputs/http/MessageProcessor.java

```diff
@@ -35,11 +44,65 @@ public void onRejection() {
     @Override
     public void run() {
         try {
-            final FullHttpResponse response = messageHandler.onNewMessage(remoteAddress, req);
+            final HttpResponse response;
+            final String token = req.headers().get(HttpHeaderNames.AUTHORIZATION);
+            if (messageHandler.validatesToken(token)) {
+                final Map<String, String> formattedHeaders = formatHeaders(req.headers());
+                final String body = req.content().toString(charset);
+                if (messageHandler.onNewMessage(remoteAddress, formattedHeaders, body)) {
+                    response = generateResponse(messageHandler.responseHeaders());
+                } else {
+                    response = generateFailedResponse(HttpResponseStatus.INTERNAL_SERVER_ERROR);
+                }
+            } else {
+                response = generateFailedResponse(HttpResponseStatus.UNAUTHORIZED);
+            }
             ctx.writeAndFlush(response);
         } finally {
             req.release();
         }
     }
+
+    private FullHttpResponse generateFailedResponse(HttpResponseStatus status) {
+        final FullHttpResponse response = new DefaultFullHttpResponse(req.protocolVersion(), status);
+        response.headers().set(HttpHeaderNames.CONTENT_LENGTH, 0);
+        return response;
+    }
+
+    private FullHttpResponse generateResponse(Map<String, String> stringHeaders) {
+
+        final ByteBuf payload = Unpooled.wrappedBuffer("ok".getBytes(charset));
+        final FullHttpResponse response = new DefaultFullHttpResponse(
+                req.protocolVersion(),
+                HttpResponseStatus.OK,
+                payload);
+
+        final DefaultHttpHeaders headers = new DefaultHttpHeaders();
+        for(String key : stringHeaders.keySet()) {
+            headers.set(key, stringHeaders.get(key));
+        }
+        response.headers().set(headers);
```

Comment:
It looks like the authorization header is now being echo'ed back on the response. We should not echo it back since it is just encoded, not encrypted. (`echo amFrZTpmb28= | base64 -D` to see my super secret un/password)

```
 "headers" => {
          "authorization" => "Basic amFrZTpmb28=",
```